solver/jobs.go

```diff
@@ -505,7 +505,36 @@ func (j *Job) Build(ctx context.Context, e Edge) (CachedResult, error) {
 		return nil, err
 	}
 	e.Vertex = v
-	return j.list.s.build(ctx, e)
+
+	res, err := j.list.s.build(ctx, e)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: returned out build info
+	_ = j.list.mergeBuildInfo(j)
+
+	return res, nil
+}
+
+func (jl *Solver) mergeBuildInfo(j *Job) map[string]string {
+	jl.mu.Lock()
+	defer jl.mu.Unlock()
+	bi := make(map[string]string)
+	for _, st := range jl.actives {
```

Comment:
the walking should still start from the `e`. Find the matching state from `jl.actives` for `e` and then walk its inputs.